modules/create_timepoint/jsx/createTimepointIndex.js

```diff
@@ -118,50 +118,46 @@ class CreateTimepoint extends React.Component {
-          state.errors = data.errors;
-          this.setState(state);
-        }
-        // Populate the select options for subproject.
-        if (data.subproject) {
-          const state = Object.assign({}, this.state);
-          state.form.options.subproject = data.subproject;
-          state.form.value.subproject = null;
-          state.form.display.subproject = true;
-          this.setState(state);
-        }
-        // Populate the select options for psc.
-        if (data.psc) {
-          const state = Object.assign({}, this.state);
-          state.form.options.psc = data.psc;
-          state.form.value.psc = null;
-          state.form.display.psc = true;
-          this.setState(state);
-        }
-        // Populate the select options for visit.
-        if (data.visit) {
-          // Store the (complete) visit selection information.
-          const state = Object.assign({}, this.state);
-          state.storage.visit = data.visit;
-          this.setState(state);
-          // Handle visit selection.
-          this.handleVisitLabel();
-        }
-        // Display form to user.
-        this.setState({isLoaded: true});
-      } else {
-        response.text().then((message) => {
-          const state = Object.assign({}, this.state);
-          state.errors = [{error: message}];
-          this.setState(state);
-        });
-      }
+    ).then((response) => response.json())
+      .then(
+        (data) => {
+          // Populate the form errors.
+          if (data.errors) {
+            const state = Object.assign({}, this.state);
+            state.errors = data.errors;
+            this.setState(state);
+          }
+          // Populate the select options for subproject.
+          if (data.subproject) {
+            const state = Object.assign({}, this.state);
+            state.form.options.subproject = data.subproject;
+            state.form.value.subproject = null;
+            state.form.display.subproject = true;
+            this.setState(state);
+          }
+          // Populate the select options for psc.
+          if (data.psc) {
+            const state = Object.assign({}, this.state);
+            state.form.options.psc = data.psc;
+            state.form.value.psc = null;
+            state.form.display.psc = true;
+            this.setState(state);
+          }
+          // Populate the select options for visit.
+          if (data.visit) {
+            // Store the (complete) visit selection information.
+            const state = Object.assign({}, this.state);
+            state.storage.visit = data.visit;
+            this.setState(state);
+            // Handle visit selection.
+            this.handleVisitLabel();
+          }
+          // Display form to user.
+          this.setState({isLoaded: true});
         }).catch((error) => {
-          console.log(error);
+      const state = Object.assign({}, this.state);
+      state.errors = [{message: error}];
+      this.setState(state);
```

Comment:
the error is going into state but does it get rendered? should at least have console.error

Reply:
The error will be displayed to the user. See line 262 and 313. It's how all the errors are shown to the user.